api/assignments/assignmentsRouter.js

```diff
@@ -18,233 +18,22 @@ router.get('/', authRequired, adminRequired, (req, res, next) => {
     });
 });
 
-// get all the mentees a mentor has by the mentor's id
-
-router.get(
-  '/mentor/:id',
-  authRequired,
-  validProfileID,
-  adminRequired,
-  (req, res, next) => {
-    const id = req.params.id;
-    Assignment.findByMentorId(id)
-      .then((mentees) => {
-        if (mentees) {
-          res.status(200).json(mentees);
-        } else {
-          res
-            .status(404)
-            .json({ error: 'Assignment Not Found, Check mentor ID' });
-        }
-      })
-      .catch((err) => {
-        next({ status: 500, message: err.message });
-      });
-  }
-);
-
-// get all the mentors a mentee has by the mentee's id
-
-router.get(
-  '/mentee/:id',
-  authRequired,
-  validProfileID,
-  adminRequired,
-  (req, res, next) => {
-    const id = req.params.id;
-    Assignment.findByMenteeId(id)
-      .then((mentors) => {
-        if (mentors) {
-          res.status(200).json(mentors);
-        } else {
-          res
-            .status(404)
-            .json({ error: 'Assignment Not Found, Check mentee ID' });
-        }
-      })
-      .catch((err) => {
-        next({ status: 500, message: err.message });
-      });
-  }
-);
-
-//get current users mentors
-
-router.get('/mymentors', authRequired, (req, res, next) => {
-  const token = req.headers.authorization;
-  const User = jwt(token);
-  Assignment.findByMenteeId(User.sub)
-    .then((mentors) => {
-      if (mentors) {
-        res.status(200).json(mentors);
-      } else {
-        res.status(404).json({ error: 'Mentors not found' });
-      }
+// get all assignments by profile ID
+
+router.get('/:role/:profile_id', (req, res, next) => {
+  const { role, profile_id } = req.params;
+  const payload = {
+    user_id: profile_id,
+    user_type: role,
+  };
+  axios
+    .post(`${baseURL}/read/match`, payload)
+    .then((dsRes) => {
+      res.json(dsRes.data.result);
```

Comment:
You sure did take out a lot of unnecessary code excellent! I also love that you did one call that can return data for both mentors and users, and potentially admin too. Nice work can't wait to use this for the FE!